### ISSUE TITLE
[com.google.fonts/check/version_bump] not working properly?

### ISSUE THREAD

**Issue by emmamarichal** [fonttools/fontbakery#4340]
## Observed behaviour

https://fonts.google.com/specimen/Heebo
Version number on the API: 3.001

I made a mistake and put 3.000 for the new upgrade, but it was not detected by fontbakery (neither locally, nor by the CI in the PR)
https://github.com/google/fonts/pull/6948

## Expected behaviour

According to this, it should be detected right? 

```
   if v_number < api_gfonts_v_number:
        passed = False
        yield FAIL, (
            f"Version number {v_number} is less than"
            f" version on Google Fonts ({api_gfonts_v_number})."
        )
```

## Resources and steps needed to reproduce

(upgrade - version number to modify) v3.000: https://github.com/OdedEzer/heebo/tree/master/fonts/variable
(current version) v3.001 https://fonts.google.com/specimen/Heebo
PR: https://github.com/google/fonts/pull/6948

**Comment by felipesanches** [comment]
yes, the FontBakery `@condition` that fetches the files from **google/fonts** repo was getting confused when building the URL due to the brackets used on a variable font name. I have now updated the implementation of that specific `condition` to handle it properly.

I also took the opportunity to improve the formatting of version numbers on the check messages. Those numbers were showing up like this:
![Screenshot from 2023-11-29 06-47-18](https://github.com/fonttools/fontbakery/assets/213676/aa67d164-226a-433b-ac7a-dbd9b0d200f5)

And now they show properly as this:
![Screenshot from 2023-11-29 07-03-02](https://github.com/fonttools/fontbakery/assets/213676/5be6f9ab-b63e-4ba4-8ed9-ab10ce32946b)
